File: backend/src/main/java/com/prova/demo/ModelloA.java
package com.prova.demo;

import java.util.ArrayList;
import java.util.Date;
import java.time.*;

import javax.persistence.Entity;
import javax.persistence.GeneratedValue;
import javax.persistence.Id;
import javax.validation.constraints.NotBlank;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

@Data
@Entity
@NoArgsConstructor
@AllArgsConstructor
public class ModelloA {

	private @Id @GeneratedValue Long codiceId;
	private @NotBlank String tipoEvento;
	private @NotBlank String luogoEvento;
	private LocalDateTime orarioArrivoEvento;
	private Date dataArrivoEvento;
	private ArrayList<Volontari> volontariImpiegati;
	private ArrayList<Automezzi> automezziImpiegati;
	private String note;
	private @NotBlank String firmaResponsabileOrg;
}

File: frontend/src/app/modelloA.ts
export class ModelloA {

    codiceId?: number;
    tipoEvento: string;
    luogoEvento: string;
    orarioArrivoEvento?: string;
    dataArrivoEvento?: string;
    note?: string;
    firmaResponsabileOrg: string;

    constructor(codiceId: number, tipoEvento: string, luogoEvento: string, orarioArrivoEvento: string, dataArrivoEvento: string, note: string, firmaResponsabileOrg: string) {
        this.codiceId = codiceId;
        this.tipoEvento = tipoEvento;
        this.luogoEvento = luogoEvento;
        this.orarioArrivoEvento = orarioArrivoEvento;
        this.dataArrivoEvento = dataArrivoEvento;
        this.note = note;
        this.firmaResponsabileOrg = firmaResponsabileOrg;
    }

}